pkg/operator/events/recorder_upstream.go

```diff
@@ -13,23 +13,38 @@ import (
 
 // NewKubeRecorder returns new event recorder.
 func NewKubeRecorder(client corev1client.EventInterface, sourceComponentName string, involvedObjectRef *corev1.ObjectReference) Recorder {
-	broadcaster := record.NewBroadcaster()
-	broadcaster.StartLogging(glog.Infof)
-	broadcaster.StartRecordingToSink(&corev1client.EventSinkImpl{Interface: client})
-	eventRecorder := broadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: sourceComponentName})
-
-	return &upstreamRecorder{
-		eventRecorder:     eventRecorder,
+	return (&upstreamRecorder{
+		client:            client,
+		component:         sourceComponentName,
 		involvedObjectRef: involvedObjectRef,
-	}
+	}).ForComponent(sourceComponentName)
```

Comment:
just set it

Reply:
it will duplicate the code from ForComponent()